connect_four.py

```diff
@@ -25,48 +29,53 @@ def print_board(self) -> None:
 
             print(print_row)
             row_count -= 1
-        
-        print("  A  B  C  D  E  F  G")  #TODO needs to flex to board size
+
+        print_row = " "
+        for lett in range(self.num_cols):
```

Comment:
Theres a much simpler, more concise, and less error prone way to do this: Take a look at this: https://www.w3schools.com/python/ref_string_join.asp. I would recommend applying it to all places where you do a join of strings in the code (earlier in the method too)